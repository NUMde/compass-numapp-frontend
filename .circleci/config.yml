# Use the latest 2.1 version of CircleCI pipeline process engine.
# See: https://circleci.com/docs/configuration-reference
# For a detailed guide to building and testing on iOS, read the docs:
# https://circleci.com/docs/testing-ios/
version: 2.1

jobs:
  node:
    working_directory: ~/compass-template
    docker: 
        - image: cimg/node:18.12
    steps: 
      - checkout

      - run: cd ./compass-numapp-template
      - restore_cache:
          key: npm-v1-{{ checksum "./compass-numapp-template/package-lock.json" }}-{{ arch }}

      - restore_cache:
          key: node-v1-{{ checksum "./compass-numapp-template/package.json" }}-{{ arch }}

      - run: cd ~ && npx react-native init compass --template ~/compass-template --title "COMPASS"
      
      - save_cache:
          key: npm-v1-{{ checksum "./compass-numapp-template/package-lock.json" }}-{{ arch }}
          paths:
            - ~/.cache/npm

      - save_cache:
          key: node-v1-{{ checksum "./compass-numapp-template/package.json" }}-{{ arch }}
          paths:
            - ./compass/node_modules
      - persist_to_workspace:
          root: ../compass
          paths:
            - '*'
  build_ios:
    working_directory: ~/compass/ios
    # execution environment.
    # See: https://circleci.com/docs/configuration-reference/#docker-machine-macos-windows-executor
    macos:
      xcode: 14.2.0
    environment:
      CI: 'true'

    # See: https://circleci.com/docs/configuration-reference/#steps
    steps:
      - attach_workspace: 
          at: ~/compass/
      - run: pod install
      - run: fastlane scan # Run tests using Fastlane

      # Collect XML test results data to show in the UI, and save the same XML
      # files under test-results folder in the Artifacts tab
      - store_test_results:
          path: test_output
      - store_artifacts:
          path: test_output
          destination: scan-output

# Invoke jobs via workflows
# See: https://circleci.com/docs/configuration-reference/#workflows
workflows:
  build: # This is the name of the workflow, feel free to change it to better match your workflow.
    # Inside the workflow, you define the jobs you want to run.
    jobs:
      - node
      - build_ios:
          requires: 
            - node

    # - build_android
    #     requires:
    #       - node
