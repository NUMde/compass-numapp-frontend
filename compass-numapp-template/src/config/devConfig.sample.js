import { name as appName } from '../../appName.json';

export default {
  // DEVELOPMENT OPTIONS
  /*-----------------------------------------------------------------------------------*/

  // these options are meant for development only.

  /** dev-option:
   * skips the QR login after 500ms */
  automateQrLogin: __DEV__ && false,

  /** dev-option:
   * subject-id for automated login */
  automateQrLoginSubjectId: `{"AppIdentifier":"${appName}","SubjectId":"7bfc3b07-a97d-4e11-8ac6-b970c1745476"}`,

  /** dev-option:
   * shows a button to erase all data (in the about-menu) - dev-only */
  showEraseAll: __DEV__ && true,

  /** dev-option:
   * shows a button to log out the user (in the about-menu) - dev-only */
  showLogout: __DEV__ && true,

  /** dev-option:
   * logs out the response-json parsed as an object in the developer console
   * */
  logPureResponse: __DEV__ && false,

  /** dev-option:
   * logs out the response-json in the developer console
   * */
  logPureResponseJSON: __DEV__ && false,

  /** dev-option:
   * logs out the encrypted response-json in the developer console
   * */
  logEncryptedResponse: __DEV__ && false,

  /** dev-option:
   * used the locally available questionnaire.js instead of the procured one from the backend */
  useLocalQuestionnaireInsteadOfTheReceivedOne: __DEV__ && false,

  /** dev-option:
   * skips the id-comparison */
  skipIncomingQuestionnaireCheck: __DEV__ && false,

  // development-base-backend-uri
  baseUri: 'http://127.0.0.1:8080/api/',
};
