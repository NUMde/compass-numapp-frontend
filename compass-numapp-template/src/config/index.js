import theme from './theme';
import text from './textConfig';
import appConfig from './appConfig';

/** combine the default values with the custom values */
export {
  /** provides variables controlling the style of the application */
  theme,
  /** provides the text-strings visible the application */
  text as strings,
  /** provides options to alter the behaviour of the app */
  appConfig,
};
