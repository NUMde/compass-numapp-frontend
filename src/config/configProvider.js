import defaultThemeValues from '../theme/theme';
import defaultTextValues from './textConfig';
import defaultAppConfigValues from './appConfig';

import defaultThemeValues from '../theme/theme';
import defaultAppConfigValues from './appConfig';

import customThemeValues from '../CUSTOMIZATION/customTheme';
import customAppConfigValues from '../CUSTOMIZATION/customAppConfig';

let exportValues  = {

    /** provides variables controlling the style of the application */
    theme: {
        ...defaultThemeValues,
        ...customThemeValues
    },

    /** provides options to alter the behaviour of the app */
    appConfig: {
        ...defaultAppConfigValues,
        ...customAppConfigValues
    },
};

/** combine the default values with the custom values */
export default exportValues
