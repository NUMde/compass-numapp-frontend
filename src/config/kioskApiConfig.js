/***********************************************************************************************
imports
***********************************************************************************************/

// the predefined questionnaire the demo is gonna use
import hardcodedTestQuestionnaire from "../assets/files/questionnaire";

/***********************************************************************************************
 constants
 ***********************************************************************************************/

 // is used to keep track of the completed transmissions (faked ones)
let playthrough = 1;

// is used to determine if a report was just sent out
let sentOutReport = false;

// is used to keep track if the first response was already sent out
let sentOutTheFirstOne = false;

// default user data
const defaultMockUserData = {
    firstTime: false,
    current_interval: 7,
    additional_iterations_left: 0,
    current_questionnaire_id: "final",
    due_date: "9999-12-30T23:00:00.000Z",
    start_date: "2021-07-15T04:00:00.000Z",
    subjectId: "7bfc3b07-a97d-4e11-8ac6-b970c1745476",
    pushAppGUID: "f98fc829-1d4d-418b-b05b-a147fdb9c0f3",
    current_instance_id: "a177e102-ecfa-490e-81a7-04cf7ebf22cf",
    pushClientSecret: "ab4b7d62-1fde-4a46-a566-2d1ada678df3",
    recipient_certificate_pem_string:
        "-----BEGIN CERTIFICATE-----\nMIIEOzCCAyOgAwIBAgIUc0iY16RQuluxpRL7hsmq/FRaWt0wDQYJKoZIhvcNAQEL\nBQAwgawxCzAJBgNVBAYTAkRFMQwwCgYDVQQIDANSTFAxDjAMBgNVBAcMBU1haW56\nMREwDwYDVQQKDAhVTSBNYWluejEeMBwGA1UECwwVUHJldmVudGl2ZSBDYXJkaW9s\nb2d5MRUwEwYDVQQDDAxHQ1NTdHVkeUNlcnQxNTAzBgkqhkiG9w0BCQEWJnN2ZW4t\nb2xpdmVyLnRyb2Vic0B1bmltZWRpemluLW1haW56LmRlMB4XDTIwMDkwMTExMjk0\nNFoXDTIzMDUyOTExMjk0NFowgawxCzAJBgNVBAYTAkRFMQwwCgYDVQQIDANSTFAx\nDjAMBgNVBAcMBU1haW56MREwDwYDVQQKDAhVTSBNYWluejEeMBwGA1UECwwVUHJl\ndmVudGl2ZSBDYXJkaW9sb2d5MRUwEwYDVQQDDAxHQ1NTdHVkeUNlcnQxNTAzBgkq\nhkiG9w0BCQEWJnN2ZW4tb2xpdmVyLnRyb2Vic0B1bmltZWRpemluLW1haW56LmRl\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnmY9Ep2SIoo824dWk1y8\ngonKArnhwVyW3AAVLpz9UtmkfElLiNLUbVOv6dGyCG9W8Y0DFwUU33EmFBlsmX3R\n7LbnwLYn7R6ntSakr58P6yNjMXY2CT2GjI0m63DPbLh6S3J6cO/8HNRnL3zW4O9B\n6FtgVg7Yf2koFjXz/kUYOFeqsPoqqLk70pO1nPGTjkz2JhBJQ+SaulTo19ZQ2tkT\nVP3zTZ9Urk8i/3fCU1TKTbYL/WwQepJlN8BwxzjIDQ2wA62nXzzlHQN98MnQGgY4\nXvj679rLFLsz2vgVmLnGPFwFL5op/6gmsKz7Q+B3zGco6qGyyL9N/D4fLKOVxnMy\nIwIDAQABo1MwUTAdBgNVHQ4EFgQUyMC1pkSWjQbIFZ/o4SEV9IqHJmwwHwYDVR0j\nBBgwFoAUyMC1pkSWjQbIFZ/o4SEV9IqHJmwwDwYDVR0TAQH/BAUwAwEB/zANBgkq\nhkiG9w0BAQsFAAOCAQEAIAAhhtVpDf/767pg8xxwaETYjPRL4pKW4H5b9t2Vr41M\nAdCUou9dbtWlqeFvY/eBby9DM05lI+D8hX2wYVRr7Pc80fuw5k4f+uTry2jc0vFV\nvHhv9sHJq3+EX/P4Nckb7M5lKI+d6GDiqzU3r/ARstoRVbQHRu2f0W7SW0pVPwlQ\nSFv61NIkjQB19SAO1IAR4Z/UrZ7Y0vnKQZ2fqFUHoD5wz4jXhHFsungocg9IxJ74\nKXaMj1nTtjCGxx11jQCNLyrVjI/XgNoPQ1v2FIQxmPJju9MAXQ2AIPA2NTXaDXHk\nYDlCrh7jxSgVWfrijhMFO/BmHDlumBpOPO1S4XG4qA==\n-----END CERTIFICATE-----",
};

/***********************************************************************************************
methods
***********************************************************************************************/

/**
 * calculates the next start date
 * @param  {boolean} laterOn tells us if we need to up the date
 */
const getStartDate = (laterOn) => {
    const now = new Date;
    now.setDate(now.getDate() + (laterOn ? 3 * playthrough : 0));
    return now.toISOString()
};

/**
 * calculates the next due date
 * @param  {boolean} laterOn tells us if we need to up the date
 */
const getDueDate = (laterOn) => {
    const then = new Date
    then.setDate(then.getDate() + (laterOn ? 10 * playthrough : 7));
    return then.toISOString()
};

/**
 * generates a fitting set of userdata
 */
const generateMockUserData = () => {
    // first time user
    if(!sentOutTheFirstOne) return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(),
        due_date: getDueDate(),
        firstTime: true,
    };
    // after sending out a regular questionnaire
    if(!sentOutReport) return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(true),
        due_date: getDueDate(true),
        firstTime: false,
    };
    // after sending out a report
    return {
        ...defaultMockUserData,
        current_interval: playthrough,
        start_date: getStartDate(),
        due_date: getDueDate(),
        firstTime: false,
    };
};

/**
 * triggers the mock for the sendQuestionnaire call and updates the status variables
 */
const sendQuestionnaire = () => {
    sentOutTheFirstOne = true;
    sentOutReport = false;
    playthrough += 1;
    return Promise.resolve(generateMockUserData());
};

/**
 * triggers the mock for the sendReport call and updates a status variable
 */
const sendReport = () => {
    sentOutReport = true;
    return Promise.resolve(generateMockUserData());
};

/***********************************************************************************************
export
***********************************************************************************************/

export default {
    sendReport,
    sendQuestionnaire,
    updateDeviceToken: () => Promise.resolve({ data: "something something" }),
    login: () => Promise.resolve({ data: generateMockUserData() }),
    getUserUpdate: () => Promise.resolve({ data: generateMockUserData() }),
    getBaseQuestionnaire: () => Promise.resolve({ data: hardcodedTestQuestionnaire })
};
