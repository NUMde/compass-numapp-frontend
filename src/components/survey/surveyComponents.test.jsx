import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import translate from '~services/localization';
import config from '~config/configProvider';

import CategoriesList from './categoriesList';

describe('categoriesList', () => {
  const categories = [
    { linkId: '1', text: 'item1' },
    { linkId: '2', text: 'item2' },
    { linkId: '3', text: 'item3' },
  ];
  const itemMap = {
    1: { done: false, started: false },
    2: { done: false, started: true },
    3: { done: true, started: true },
  };

  const showQuestionnaireModal = jest.fn();

  it('should render the categories list', () => {
    const tree = render(
      <CategoriesList
        categories={categories}
        itemMap={itemMap}
        showQuestionnaireModal={showQuestionnaireModal}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should open the modal when clicking category', () => {
    const { getByText } = render(
      <CategoriesList
        categories={categories}
        itemMap={itemMap}
        showQuestionnaireModal={showQuestionnaireModal}
      />,
    );
    fireEvent.press(getByText(categories[0].text));
    expect(showQuestionnaireModal).toHaveBeenCalledWith(0);
    fireEvent.press(getByText(categories[1].text));
    expect(showQuestionnaireModal).toHaveBeenCalledWith(1);
  });

  it('a11y hints should match', () => {
    const { getByA11yLabel } = render(
      <CategoriesList
        categories={categories}
        itemMap={itemMap}
        showQuestionnaireModal={showQuestionnaireModal}
      />,
    );

    expect(getByA11yLabel(categories[0].text)).toBeTruthy();
    expect(getByA11yLabel(categories[0].text).props.accessibilityHint).toBe(
      translate('accessibility').questionnaire.categoryCellHint +
        translate('accessibility').questionnaire.category +
        translate('accessibility').questionnaire.notStarted,
    );

    expect(getByA11yLabel(categories[1].text)).toBeTruthy();
    expect(getByA11yLabel(categories[1].text).props.accessibilityHint).toBe(
      translate('accessibility').questionnaire.categoryCellHint +
        translate('accessibility').questionnaire.category +
        translate('accessibility').questionnaire.notFinished,
    );

    expect(getByA11yLabel(categories[2].text)).toBeTruthy();
    expect(getByA11yLabel(categories[2].text).props.accessibilityHint).toBe(
      translate('accessibility').questionnaire.categoryCellHint +
        translate('accessibility').questionnaire.category +
        translate('accessibility').questionnaire.finished,
    );
  });

  it('chevrons should match', () => {
    const { getByTestId } = render(
      <CategoriesList
        categories={categories}
        itemMap={itemMap}
        showQuestionnaireModal={showQuestionnaireModal}
      />,
    );

    expect(
      getByTestId(`${categories[0].linkId}_icon`).props.children.props.style
        .backgroundColor,
    ).toBe(config.theme.colors.alert);

    expect(
      getByTestId(`${categories[1].linkId}_icon`).props.children.props.style
        .backgroundColor,
    ).toBe(config.theme.colors.secondary);
    expect(
      getByTestId(`${categories[2].linkId}_icon`).props.children.props.style
        .backgroundColor,
    ).toBe(config.theme.colors.success);
  });
});
