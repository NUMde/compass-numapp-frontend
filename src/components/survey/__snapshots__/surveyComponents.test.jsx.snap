// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categoriesList should render the categories list 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F9FA",
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityHint="Opens modal to fill out this questionnaire entry.This Category has not been answered yet."
    accessibilityLabel="item1"
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F9FA",
          "borderBottomColor": "#F8F9FA",
          "borderBottomWidth": 1,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
          "width": "100%",
        }
      }
      testID="padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1F2532",
              "fontFamily": "IBMPlexSans-SemiBold",
              "fontSize": 30,
              "lineHeight": 37.5,
            }
          }
          testID="listItemTitle"
        >
          item1
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 16,
            "height": 28,
            "margin": 7,
            "overflow": "hidden",
            "width": 28,
          }
        }
      >
        <View
          testID="1_icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EB576A",
                "borderRadius": 16,
                "height": 28,
                "justifyContent": "center",
                "width": 28,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 12,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ó°²¶
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityHint="Opens modal to fill out this questionnaire entry.This Category has already been started, but isn't completely filled in yet."
    accessibilityLabel="item2"
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F9FA",
          "borderBottomColor": "#F8F9FA",
          "borderBottomWidth": 1,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
          "width": "100%",
        }
      }
      testID="padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1F2532",
              "fontFamily": "IBMPlexSans-SemiBold",
              "fontSize": 30,
              "lineHeight": 37.5,
            }
          }
          testID="listItemTitle"
        >
          item2
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 16,
            "height": 28,
            "margin": 7,
            "overflow": "hidden",
            "width": 28,
          }
        }
      >
        <View
          testID="2_icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFC107",
                "borderRadius": 16,
                "height": 28,
                "justifyContent": "center",
                "width": 28,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 12,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ó°
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityHint="Opens modal to fill out this questionnaire entry.This Category is fully complete."
    accessibilityLabel="item3"
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F8F9FA",
          "borderBottomColor": "#F8F9FA",
          "borderBottomWidth": 1,
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
          "width": "100%",
        }
      }
      testID="padView"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#1F2532",
              "fontFamily": "IBMPlexSans-SemiBold",
              "fontSize": 30,
              "lineHeight": 37.5,
            }
          }
          testID="listItemTitle"
        >
          item3
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 16,
            "height": 28,
            "margin": 7,
            "overflow": "hidden",
            "width": 28,
          }
        }
      >
        <View
          testID="3_icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#006864",
                "borderRadius": 16,
                "height": 28,
                "justifyContent": "center",
                "width": 28,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontSize": 12,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              ó°¬
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
